ark/phenotyping/cell_cluster_utils.py

```diff
@@ -649,47 +678,49 @@ def cell_consensus_cluster(fovs, channels, base_dir, cell_som_cluster_cols, max_
     )
 
     # compute the weighted channel average expression per cell SOM cluster
-    print("Compute average weighted channel expression across cell SOM clusters")
-    cell_som_cluster_channel_avg = compute_cell_cluster_channel_avg(
-        fovs,
-        channels,
-        base_dir,
-        weighted_cell_channel_name,
-        cluster_counts_size_norm_name,
-        'cell_som_cluster'
-    )
+    # NOTE: this will only be applicable for pixel cluster expression columns
+    if weighted_cell_channel_name:
+        print("Compute average weighted channel expression across cell SOM clusters")
+        cell_som_cluster_channel_avg = compute_cell_cluster_channel_avg(
+            fovs,
+            channels,
+            base_dir,
+            weighted_cell_channel_name,
+            cluster_counts_size_norm_name,
+            'cell_som_cluster'
+        )
 
-    # merge metacluster assignments into cell_som_cluster_channel_avg
-    print(
-        "Mapping meta cluster values onto average weighted channel expression"
-        "across cell SOM clusters"
-    )
-    cell_som_cluster_channel_avg = pd.merge_asof(
-        cell_som_cluster_channel_avg, cell_cc.mapping, on='cell_som_cluster'
-    )
+        # merge metacluster assignments into cell_som_cluster_channel_avg
+        print(
+            "Mapping meta cluster values onto average weighted channel expression"
+            "across cell SOM clusters"
+        )
+        cell_som_cluster_channel_avg = pd.merge_asof(
+            cell_som_cluster_channel_avg, cell_cc.mapping, on='cell_som_cluster'
+        )
 
-    # save the weighted channel average expression per cell cluster
-    cell_som_cluster_channel_avg.to_csv(
-        os.path.join(base_dir, cell_som_cluster_channel_avg_name),
-        index=False
-    )
+        # save the weighted channel average expression per cell cluster
+        cell_som_cluster_channel_avg.to_csv(
+            os.path.join(base_dir, cell_som_cluster_channel_avg_name),
+            index=False
+        )
 
-    # compute the weighted channel average expression per cell meta cluster
-    print("Compute average weighted channel expression across cell meta clusters")
-    cell_meta_cluster_channel_avg = compute_cell_cluster_channel_avg(
-        fovs,
-        channels,
-        base_dir,
-        weighted_cell_channel_name,
-        cluster_counts_size_norm_name,
-        'cell_meta_cluster'
-    )
+        # compute the weighted channel average expression per cell meta cluster
+        print("Compute average weighted channel expression across cell meta clusters")
```

Comment:
I think part of the reason the code complexity is high comes from the clustering functions also including the pre- and post-processing. If you think it'll create more headaches, let me know, but to me it seems like things would be simpler if these steps were pulled out into separate functions. For example, computing the averages across cells.

Reply:
Yeah I have an issue in #853 to address this. We can address that here.